specs/_features/eip7547/beacon-chain.md

```diff
@@ -0,0 +1,294 @@
+# EIP-7547 -- The Beacon Chain
+
+## Table of contents
+
+<!-- TOC -->
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Introduction](#introduction)
+- [Constants](#constants)
+  - [Domain types](#domain-types)
+- [Preset](#preset)
+  - [Execution](#execution)
+- [Containers](#containers)
+  - [New Containers](#new-containers)
+    - [`SignedInclusionListSummary`](#signedinclusionlistsummary)
+  - [Execution engine](#execution-engine)
+    - [Request data](#request-data)
+      - [New `NewInclusionListRequest`](#new-newinclusionlistrequest)
+    - [Engine APIs](#engine-apis)
+    - [New `notify_new_inclusion_list`](#new-notify_new_inclusion_list)
+  - [Extended containers](#extended-containers)
+    - [`ExecutionPayload`](#executionpayload)
+    - [`ExecutionPayloadHeader`](#executionpayloadheader)
+    - [`BeaconState`](#beaconstate)
+  - [Block processing](#block-processing)
+    - [Block header](#block-header)
+    - [Execution payload](#execution-payload)
+      - [Modified `process_execution_payload`](#modified-process_execution_payload)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- /TOC -->
+
+## Introduction
+
+This is the beacon chain specification to add an inclusion list mechanism to allow forced transaction inclusion. Refers to [EIP-7547](https://eips.ethereum.org/EIPS/eip-7547).
+
+*Note:* This specification is built upon [Deneb](../../deneb/beacon_chain.md) and is under active development.
+
+## Constants
+
+### Domain types
+
+| Name | Value |
+| - | - |
+| `DOMAIN_INCLUSION_LIST_SUMMARY`     | `DomainType('0x0B000000')` |
+
+## Preset
+
+### Execution
+
+| Name | Value |
+| - | - |
+| `MAX_TRANSACTIONS_PER_INCLUSION_LIST` |  `uint64(256)` |
```

Comment:
I'd consider having this number being much larger, something like 2**14 or so, to be future proof and not change the SSZ hashing of this.

Reply:
done!